requests.go

```diff
@@ -46,6 +50,9 @@ type HTTPClient struct {
 	errorHandler  func(io.Reader) error // default error handler for all requests returning unexpected status
 	bodyHandler   BodyHandlerFunc       // default body handler for requests (when Content-Type is not automatically handled by this library)
 	logger        *zap.Logger           // logger to use (only debug messages are printed by the library; defaults to noop logger)
+	certPEMBlock  []byte
+	keyPEMBlock   []byte
+	caCert        []byte
```

Comment:
How can these attributes be set? I don't see that they are exposed in any way.

I think you should add a method on the HTTPClient struct, maybe something like `WithCertificate(certPEMBlock, keyPEMBlock, caCert)` that sets these attributes. Then you provide these to `DefaultTransport` (together with `noTLSVerify`) in order to generate the correct `tls.Config` struct. There's no reason for the `sslAuth` constant. The `authType` type is for authentication methods (Basic, Digest, 2-factor), this isn't an authentication method.